From ea706b389fd896318bd7310b058fedf11d1b6758 Mon Sep 17 00:00:00 2001
From: Koen Vandeputte <koen.vandeputte@citymesh.com>
Date: Fri, 7 Jul 2023 16:48:26 +0200
Subject: [PATCH 19/28] iw: fix attribute size mismatch

NL80211_ATTR_MAX_AP_ASSOC_STA gets packed as u32 in the kernel.
Change the receiving side to match this, or it will be wrong
on big-endian.

Signed-off-by: Koen Vandeputte <koen.vandeputte@citymesh.com>
Link: https://lore.kernel.org/r/20230707144826.3043151-1-koen.vandeputte@citymesh.com
Signed-off-by: Johannes Berg <johannes.berg@intel.com>
---
 info.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/info.c b/info.c
index 364f9b3..4c5f463 100644
--- a/info.c
+++ b/info.c
@@ -879,7 +879,7 @@ broken_combination:
 
 	if (tb_msg[NL80211_ATTR_MAX_AP_ASSOC_STA])
 		printf("\tMaximum associated stations in AP mode: %u\n",
-		       nla_get_u16(tb_msg[NL80211_ATTR_MAX_AP_ASSOC_STA]));
+		       nla_get_u32(tb_msg[NL80211_ATTR_MAX_AP_ASSOC_STA]));
 
 	return NL_SKIP;
 }
-- 
2.39.2

